### ISSUE TITLE
Remove floaters

### ISSUE THREAD

**Issue by pierotofy** [pierotofy/OpenSplat#57]
There should be a way to reduce/filter floaters.

![image](https://github.com/pierotofy/OpenSplat/assets/1951843/2d06d4c6-2392-4a50-99ff-54e8a9bbe7e5)

These significantly reduce the quality of visualization.

I thought about doing a nearest neighbor search from of the splat means compared to the points in the initial point cloud and filter any splat that is not within X units of an original point. But I don't know if this would work, as certain areas could have no initialization points. If there are better/proven methods I'd like to hear it.

**Comment by pfxuan** [comment]
[This](https://github.com/francescofugazzi/3dgsconverter/blob/37dc41e969db2c041d3a815ab0692552a80d236d/gsconverter/utils/base_converter.py#L97) method can give some help. But it’s a little computation heavy.

**Comment by pierotofy** [comment]
Or perhaps a simple statistical filter to be run at intervals, e.g. like this one from PDAL: https://pdal.io/en/2.7.0/stages/filters.outlier.html#statistical-method

**Comment by pfxuan** [comment]
Good idea. Open3D also has a few similar built-in functions: https://www.open3d.org/docs/latest/tutorial/Advanced/pointcloud_outlier_removal.html

**Comment by pierotofy** [comment]
I've tried to implement this method in https://github.com/pierotofy/OpenSplat/tree/filter, but although there's a reduction of noise, there's also an increase in artifacts (at least in the train dataset) near the object of interest. The validation loss also seems to increase.

**Comment by pfxuan** [comment]
Perhaps removing floaters could be considered as a post-processing step after the 3DGS model gets fully trained. Additionally, I found enhancing the training code, camera pose, and blurry image could substantially minimize floating artifacts.
